e2e-tests/suite/template.ts

```diff
@@ -0,0 +1,173 @@
+import { afterAll, beforeAll, describe, expect, it } from '@jest/globals';
+import { DeveloperHubClient } from "../API/developerHub-client";
+import { GitHubClient } from "../API/git/github";
+import { KubeClient } from "../API/kube";
+import { ApplicationInfo, DeploymentInfo, RepositoryInfo, ExistingModelSecret, isApplicationDeployment } from "../API/types";
+import { AITemplate } from './types';
+import { GitLabClient } from '../API/git/gitlab';
+import { GitClient } from '../API/git/types';
+
+export const generateComponentName = (template: AITemplate) => {
+  const characters = 'abcdefghijklmnopqrstuvwxyz';
+  const extraChar = characters.charAt(Math.floor(Math.random() * characters.length));
+  return `${template}-${extraChar}${Date.now()}`;
+}
+
+function hasSecret(object: ApplicationInfo): object is ExistingModelSecret {
+  return (object as ExistingModelSecret).includeModelEndpointSecret !== undefined;
+}
+
+export const templateSuite = (template: AITemplate, appInfo: ApplicationInfo, repoInfo: RepositoryInfo, deploymentInfo: DeploymentInfo, rhdhNamespace: string) => {
```

Comment:
Do we plan on introducing test cases besides the happy path in the future?

Reply:
Not sure for now. It should be pretty simple to add new cases or entire templates. These things tend to get out of hand quite quickly though, time and resource wise.

Let's first evaluate how the happy path is doing.